### PR TITLE
Feature/issue 28 create change avatar modal

### LINKED ISSUE #28: Create change avatar modal

### PR DESCRIPTION
**Pull request by boris-lazarevic**
- Add reusable Modal
- Add mocked Avatar
- Use Avatar Modal to change the avatar component

*EDIT:* @boris-lazarevic 
do not merge until `feature/issue-35_global-scripts-prisma` is merged into `feature/issue-28_create-change-avatar-modal` 🙏 🙅‍♂️ 🛑 
 
Closes #28 #35

### CODE REVIEW

File: libs/shared/ui/src/components/avatar-modal/AvatarModal.styled.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  display: grid;
+  place-items: center;
+  background: ${Color.white};
+  padding: ${space.x8};
+  border: 2px solid ${Color.dodgerblue};
+  border-radius: 5px;
+  z-index: 2;
+`;
+
+export const AvatarTitle = styled.h1``;
```
Comment (NemanjaVucic) [inline libs/shared/ui/src/components/avatar-modal/AvatarModal.styled.tsx:21]: If we do not gonna use probably should be removed.

File: libs/shared/ui/src/components/avatar-modal/AvatarModal.styled.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  border: ${({ $isSelected }) => ($isSelected ? `4px solid ${Color.dodgerblue}` : `2px solid ${Color.gray}`)};
+  transition: 0.35s ease-in-out;
+
+  &:hover {
+    transform: scale(1.6);
+    cursor: pointer;
+  }
+`;
+
+export const AvatarButtonWrapper = styled(CardButtonWrapper)``;
```
Comment (NemanjaVucic) [inline libs/shared/ui/src/components/avatar-modal/AvatarModal.styled.tsx:41]: Same as comment above ☝️

File: libs/shared/ui/src/components/avatar-modal/AvatarModal.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  showModal,
+  setShowModal,
+  selectedId,
+  setSelectedId,
+  appliedId,
+  onApplyId,
+}: AvatarModalProps) => {
+  const handleApply = () => {
+    onApplyId(selectedId);
+    setShowModal(false);
```
Comment (NemanjaVucic) [inline libs/shared/ui/src/components/avatar-modal/AvatarModal.tsx:27]: `setShowModal` comes from the parent setter function, when we have these types of callbacks we rather use the name e.g. `onShowModal`.
`setShowModal` should be called that way in cmp where has been defined.
So, just a naming issue.

Comment (NemanjaVucic) [reply libs/shared/ui/src/components/avatar-modal/AvatarModal.tsx:27]: The same applies to `setSelectedId`.

File: libs/shared/ui/src/components/modal/Modal.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  ModalTitle,
+  ApplyButton,
+  CancelButton,
+} from './Modal.styled';
+
+interface ModalProps {
+  title: string;
+  children: React.ReactNode;
+  isShowing: boolean;
+  setIsShowing: (isShowing: boolean) => void;
```
Comment (NemanjaVucic) [inline libs/shared/ui/src/components/modal/Modal.tsx:17]: Same as comments above ☝️ (naming)

File: libs/shared/ui/src/components/modal/Modal.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        <>
+          <ModalOverlay onClick={handleOverlayClick} />
+          <ModalWrapper>
+            <ModalTitle>{title}</ModalTitle>
+            <ModalBody>{children}</ModalBody>
+            <ModalButtonWrapper>
+              <ApplyButton onClick={handleApply}>Apply</ApplyButton>
+              <CancelButton onClick={handleCancel}>Cancel</CancelButton>
+            </ModalButtonWrapper>
+          </ModalWrapper>
```
Comment (NemanjaVucic) [inline libs/shared/ui/src/components/modal/Modal.tsx:52]: Beautiful 💯 🥇

**Review by NemanjaVucic — CHANGES_REQUESTED** [review]
Very good job 👌 
A few comments regarding naming, business logic is neat 🚀

File: libs/shared/ui/src/components/avatar-modal/AvatarModal.styled.tsx (hunk 2 above)
Comment (boris-lazarevic) [reply libs/shared/ui/src/components/avatar-modal/AvatarModal.styled.tsx:41]: Very nice catch. I forgot to remove unnecessary styled components when I was creating a reusable Modal. There are also a few more components that need to be removed.

File: libs/shared/ui/src/components/modal/Modal.tsx (hunk 5 above)
Comment (boris-lazarevic) [reply libs/shared/ui/src/components/modal/Modal.tsx:52]: Thank you!

**Review by NemanjaVucic — DISMISSED** [review]
Nice 👍 
In the next iteration(new issue), I would like to see CreateCard => CreatePage & connecting with BE regarding creation of user.
🚀

**Review by NemanjaVucic — APPROVED** [review]

**Review by NemanjaVucic — APPROVED** [review]

**Review by NemanjaVucic — APPROVED** [review]

**Review by NemanjaVucic — DISMISSED** [review]
Come onn

**Review by NemanjaVucic — APPROVED** [review]

**Pull request merged**